core/spec/models/spree/adjustable/adjuster/promo_spec.rb

```diff
@@ -0,0 +1,194 @@
+require 'spec_helper'
+
+describe Spree::Adjustable::Adjuster::Promo, :type => :model do
+  let(:order) { create :order_with_line_items, line_items_count: 1 }
+  let(:line_item) { order.line_items.first }
+
+  let(:subject) { Spree::Adjustable::AdjustmentsUpdater.new(line_item) }
+  let(:order_subject) { Spree::Adjustable::AdjustmentsUpdater.new(order) }
+
+  context "best promotion is always applied" do
+    let(:calculator) { Spree::Calculator::FlatRate.new(preferred_amount: 10) }
+
+    let(:source) { Spree::Promotion::Actions::CreateItemAdjustments.create calculator: calculator }
+
+    def create_adjustment(label, amount)
+      create(:adjustment, order:      order,
+             adjustable: line_item,
+             source:     source,
+             amount:     amount,
+             state:      "closed",
+             label:      label,
+             mandatory:  false)
+    end
+
+    it "should make all but the most valuable promotion adjustment ineligible, " +
+         "leaving non promotion adjustments alone" do
+      create_adjustment("Promotion A", -100)
+      create_adjustment("Promotion B", -200)
+      create_adjustment("Promotion C", -300)
+      create(:adjustment, order:  order,
+             adjustable:  line_item,
+             source:  nil,
+             amount:  -500,
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Unnecessary spacing detected.